cmdstanpy/utils.py

```diff
@@ -1431,3 +1432,27 @@ def __enter__(self) -> Tuple[str, bool]:
     def __exit__(self, exc_type, exc_val, exc_tb) -> None:  # type: ignore
         if self._tmpdir:
             shutil.rmtree(self._tmpdir, ignore_errors=True)
+
+
+def write_stan_file(code: str, directory: Union[str, None] = None) -> str:
+    """
+    Write the model code to a file and return the path based on the code hash.
+
+    :param code: Code to write to a file.
+    :param directory: Directory to write the code to (defaults to the cmdstanpy
+        models directory).
+
+    :return: Path to the file containing the code..
+    """
+    # Generate the path based on the code and check whether it already exists.
+    directory = directory or os.path.join(cmdstan_path(), 'models')
```

Comment:
I don't know if I like the idea of a cmdstan models directory. Some immediate questions/concerns come to mind:

Does it get deleted if cmdstan is rebuilt? When installing a new version of cmdstan, all older models are left behind. I as the user may not actually know where my cmdstan folder is, and unless I'm printing the return of this function won't necessarily know that they're stored here.

IMO, we should either require the directory parameter or initialize it to `"."` (the current working directory). Alternatively, we could write all these files to the tempdir, if we were okay with re-compiling each time

Reply:
Makes sense. Another option could be to use `.cmdstanpy` as other packages do, e.g. `.coverage`, `.mypy_cache`, or `.pytest_cache`.

Reply:
My philosophy is we should stray away from hiding these files in a dot folder. If users want to use it to be quick, they can, but we shouldn't hide the fact that they're doing it